site/en/docs/extensions/reference/offscreen/index.md

```diff
@@ -46,11 +46,50 @@ chrome.offscreen.createDocument({
 chrome.offscreen.closeDocument()
 ```
 
-For a complete extension, see the [offscreen-clipboard demo][gh-offscreen-clipboard] on GitHub.
+The following example shows how to check for existing offscreen documents. The `hasOffscreenDocument` method uses [clients.matchAll()](https://developer.mozilla.org/docs/Web/API/Clients/matchAll) to find existing offscreen documents.
 
-[api-runtime]: /docs/extensions/reference/runtime/
-[api-windows]: /docs/extensions/reference/windows/
-[doc-manifest]: /docs/extensions/mv3/manifest/
-[gh-offscreen-clipboard]: https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/functional-samples/cookbook.offscreen-clipboard-write
-[offscreen-reason]: /docs/extensions/reference/offscreen/#type-Reason
+```js
+async function hasOffscreenDocument(path) {
+  // Check all windows controlled by the service worker if one 
+  // of them is the offscreen document with the given path
+  const offscreenUrl = chrome.runtime.getURL(path);
+  const matchedClients = await clients.matchAll();
+  for (const client of matchedClients) {
+    if (client.url === offscreenUrl) {
+      return true;
+    }
+  }
+  return false;
+}
+```
+
+Before trying to create a new offscreen document, use `hasOffscreenDocument` to make sure that there is no existing offscreen document, as demonstrated in the following example. 
```

Comment:
Do you think we need to clarify that this wouldn't work if you had two different paths for different use cases?

Reply:
I was wondering about this as well, the problem is that when you're using different docs this become quickly very complex and there is no good default approach for handling this. I've added a sentence mentioning that we assume only a single document.